src/generic/problem.cc

```diff
@@ -8359,6 +8359,60 @@ namespace oomph
     }
   }
 
+  //==================================================================
+  /// Solve the adjoint eigenproblem
+  //==================================================================
+  void Problem::solve_adjoint_eigenproblem(
+    const unsigned& n_eval,
+    Vector<std::complex<double>>& eigenvalue,
+    Vector<DoubleVector>& eigenvector,
+    const bool& steady)
+  {
+    // If the boolean flag is steady, then make all the timesteppers steady
+    // before solving the eigenproblem. This will "switch off" the
+    // time-derivative terms in the jacobian matrix
+    if (steady)
```

Comment:
I agree that `steady` is a bit odd but `solve_steady_eigenproblem` is even odder since it makes no sense to solve an unsteady eigenvalue problem. Discuss with Andrew...

Reply:
The "steady" flag (true) by default, as it says is to ensure that the Jacobian is calculated correctly for standard eigenproblems arising from stability analysis. In some case, however, one may want to caculate the eigenvalues associated with the Jacobian for the unsteady nonlinear problem being solved, e.g. numerical stability, so it's there to give that option. The most descriptive name would be "make_timesteppers_steady"  or "set_all_timesteppers_to_steady".

Reply:
Changed to make_timesteppers_steady.